WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -360,6 +381,25 @@ private extension DashboardViewModel {
             .assign(to: &$showBlazeCampaignView)
     }
 
+    func configureOrdersResultController() {
+        ordersResultsController.onDidChangeContent = { [weak self] in
+            self?.updateResults()
+        }
+        ordersResultsController.onDidResetContent = { [weak self] in
+            self?.updateResults()
+        }
+
+        do {
+            try ordersResultsController.performFetch()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
```

Comment:
❓ @itsmeichigo   When I remove this `do... catch` part, everything works fine still. 

I believe this is because `loadHasOrdersStatus()` calls `OrderAction.checkIfStoreHasOrders()`, which does a db fetch and then network fetch. But I'm not 100% sure if my understanding is correct, so I do not remove it yet. What do you think?

Reply:
Now that I think of it, it doesn't seem ideal to update `hasOrders` in two places: `updateResults()` and `updateHasOrdersStatus()` On the other hand I do want to `updateHasOrdersStatus()` since it does a network call to fetch orders if necessary.

Do you have suggestions?

Reply:
> I believe this is because loadHasOrdersStatus() calls OrderAction.checkIfStoreHasOrders(), which does a db fetch which then triggers the ordersResultsController.onDidChangeContent() that updates hasOrders.

This is not correct. Removing the fetch makes the `ordersResultsController` redundant and `updateResults` is never called. You are updating `hasOrders` in `updateHasOrdersStatus`, that's why removing the db fetch doesn't affect `hasOrders`.

My suggestion was to update `hasOrders` only based on storage data. This means:
- Remove `updateHasOrdersStatus`.
- Rename `loadHasOrdersStatus` to `syncOrders` and make it return void. Then instead of dispatching `checkIfStoreHasOrders`, use `synchronizeOrders` instead. The goal is to update the storage with the store's order and do nothing after the sync completes.
- Keep the db fetch in `configureOrdersResultController`, when there's any change to storage, `hasOrders` will be updated thanks to the `onDidChangeContent` and `onDidResetContent` callbacks.

LMK if you have any other questions.

Reply:
Thanks! using `synchronizeOrders` here I think is a great idea that I hadn't considered before, and solely depending on storage change for updating `hasOrders` simplifies things. I refactored things that way in  https://github.com/woocommerce/woocommerce-ios/pull/12521/commits/59a939c1c64aa3a5e4be1f8f0bb765c9eb1d932f